src/main/kotlin/com/openlattice/collections/CollectionTemplates.kt

```diff
@@ -0,0 +1,33 @@
+package com.openlattice.collections
+
+
+import com.google.common.collect.Maps
+import java.util.UUID
+
+/**
+ * A Hazelcast-serializable wrapper around a MutableMap<UUID, MutableMap<UUID, UUID>>
+ * for processing entity set collection templates.
+ *
+ * @param templates A map from entity set collection ids to their templates,
+ * where their templates are maps from collection template type ids to entity set ids
+ */
+data class CollectionTemplates(
```

Comment:
This should delegate to a ConcurrentMap and most of the code in this class disappears.